subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -302,6 +308,87 @@ static void handle_wifi_twt_event(struct net_mgmt_event_callback *cb)
 	}
 }
 
+static void handle_wifi_ap_enable_result(struct net_mgmt_event_callback *cb)
+{
+	const struct wifi_status *status =
+		(const struct wifi_status *)cb->info;
+
+	if (status->status) {
+		print(context.sh, SHELL_WARNING,
+		      "AP enable request failed (%d)\n", status->status);
+	} else {
+		print(context.sh, SHELL_NORMAL, "AP enabled\n");
+	}
+}
+
+static void handle_wifi_ap_disable_result(struct net_mgmt_event_callback *cb)
+{
+	const struct wifi_status *status =
+		(const struct wifi_status *)cb->info;
+	struct wifi_ap_sta_info *sta, *next;
+
+	if (status->status) {
+		print(context.sh, SHELL_WARNING,
+		      "AP disable request failed (%d)\n", status->status);
+	} else {
+		print(context.sh, SHELL_NORMAL, "AP disabled\n");
+	}
+
+	k_mutex_lock(&wifi_ap_sta_list_lock, K_FOREVER);
+	SYS_SLIST_FOR_EACH_CONTAINER_SAFE(&wifi_ap_sta_list, sta, next, node) {
+		sys_slist_find_and_remove(&wifi_ap_sta_list, &sta->node);
+		k_free(sta);
+	}
+	k_mutex_unlock(&wifi_ap_sta_list_lock);
+}
+
+static void handle_wifi_ap_sta_connected(struct net_mgmt_event_callback *cb)
+{
+	const struct wifi_ap_sta_info *sta_info =
+		(const struct wifi_ap_sta_info *)cb->info;
+	uint8_t mac_string_buf[sizeof("xx:xx:xx:xx:xx:xx")];
+	struct wifi_ap_sta_info *sta_info_tmp = NULL;
+
+	print(context.sh, SHELL_NORMAL, "Station connected: %s\n",
+	      net_sprint_ll_addr_buf(sta_info->mac, WIFI_MAC_ADDR_LEN,
+				     mac_string_buf, sizeof(mac_string_buf)));
+	print(context.sh, SHELL_NORMAL, "====================================\n");
+	print(context.sh, SHELL_NORMAL, "Link mode: %s\n",
+	      wifi_link_mode_txt(sta_info->link_mode));
+	print(context.sh, SHELL_NORMAL, "TWT: %s\n",
+	      sta_info->twt_capable ? "Supported" : "Not supported");
+
+	k_mutex_lock(&wifi_ap_sta_list_lock, K_FOREVER);
+	sta_info_tmp = k_malloc(sizeof(struct wifi_ap_sta_info));
```

Comment:
As I had guessed, this isn't ideal as some platforms don't have the HEAP. So, will remove the HEAP usage.